stingray/gti.py

```diff
@@ -113,12 +147,72 @@ def create_gti_mask(time, gtis, safe_interval=0, min_length=0,
         fraction of dt that is tolerated at the borders of a GTI
     """
     import collections
+    import copy
+
+    try:
+        from numba import jit
+    except ImportError:
+        return create_gti_mask_complete(time, gtis,
+                                        safe_interval=safe_interval,
+                                        min_length=min_length,
+                                        return_new_gtis=return_new_gtis,
+                                        dt=dt, epsilon=epsilon)
+    gtis = np.array(gtis, dtype=np.longdouble)
+    check_gtis(gtis)
+
+    dt = assign_value_if_none(dt, np.median(np.diff(time)))
+
+    mask = np.zeros(len(time), dtype=bool)
+
+    if not isinstance(safe_interval, collections.Iterable):
+        safe_interval = np.array([safe_interval, safe_interval])
+    gti_mask = np.zeros(len(gtis), dtype=bool)
+    gtis_new = copy.deepcopy(gtis)
+    gtis_new[:, 0] = gtis[:, 0] + safe_interval[0] + dt / 2 - epsilon*dt
+    gtis_new[:, 1] = gtis[:, 1] - safe_interval[1] - dt / 2 + epsilon*dt
+    mask, gtimask = \
+        create_gti_mask_jit((time - time[0]).astype(np.float64),
+                            (gtis_new - time[0]).astype(np.float64),
+                            mask, gti_mask=gti_mask, min_length=min_length)
+    if return_new_gtis:
+        return mask, gtis[gtimask]
+    return mask
+
+
+def create_gti_mask_complete(time, gtis, safe_interval=0, min_length=0,
+                             return_new_gtis=False, dt=None, epsilon=0.001):
+    """Create GTI mask, allowing for non-constant dt.
+
+    Assumes that no overlaps are present between GTIs
+
+    Parameters
+    ----------
+    time : float array
+    gtis : [[g0_0, g0_1], [g1_0, g1_1], ...], float array-like
+
+    Returns
+    -------
+    mask : boolean array
+    new_gtis : Nx2 array
+
+    Other parameters
+    ----------------
+    safe_interval : float or [float, float]
+        A safe interval to exclude at both ends (if single float) or the start
+        and the end (if pair of values) of GTIs.
+    min_length : float
+    return_new_gtis : bool
+    dt : float
+    epsilon : float
+        fraction of dt that is tolerated at the borders of a GTI
+    """
+    import collections
 
     check_gtis(gtis)
 
     dt = assign_value_if_none(dt,
                               np.zeros_like(time) +
-                              np.median(np.diff(time)) / 2)
+                              np.median(np.diff(time)))
```

Comment:
Just curious why that `/2` got removed here. Was that a typo before?

Reply:
There were conflicting definitions of dt, and that was error-prone (divide by two later or not?). I figured it would have been better to just use the same definition used in light curve.